torch/serialization.py

```diff
@@ -312,6 +322,10 @@ def persistent_id(obj):
                     location,
                     obj.size(),
                     view_metadata)
+        # See Note [Serialize XLA tensors]
+        elif isinstance(obj, torch.Tensor) and obj.device.type == 'xla':
+            values = obj.cpu().tolist()
```

Comment:
This looks extremely wrong (even if, maybe it is accidentally doing what you want). C.f. the docs for `persistent_id`:

> For the benefit of object persistence, the pickle module supports the notion of a reference to an object outside the pickled data stream. Such objects are referenced by a persistent ID, which should be either a string of alphanumeric characters (for protocol 0) 5 or just an arbitrary object (for any newer protocol).

What are you doing here? You're reading out all of the values from the XLA tensor, and using that to form the ID. That's hardly an ID.

What I /expected/ to happen is that `__reduce_ex__` on Tensor detect when it's the XLA, decompose the tensor into its constituent pieces, and then define a rebuild function to rebuild them. What you seem to have done in the corresponding parts of this patch are just save the XLA tensor directly. Ummmm.